File: frontend/src/types/finance.ts
export interface CurrencyAmount {
  currency: string;
  amount: number;
}

export interface KpiCard {
  current: CurrencyAmount[];
  previous: CurrencyAmount[];
  change_percent: number | null;
}

export interface OrderCountCard {
  current: number;
  previous: number;
  change_percent: number | null;
}

export interface TimeSeriesPoint {
  date: string;
  currency: string;
  revenue: number;
  net_profit: number;
}

export interface DiagnosticInfo {
  orders_missing_cost: number;
  total_orders_in_period: number;
  orders_with_computed_cost: number;
}

export interface ShopFinanceResponse {
  shop_id: string;
  shop_name: string;
  period_start_iso: string;
  period_end_iso: string;
  granularity: 'day' | 'month';
  revenue: KpiCard;
  cogs: KpiCard;
  fees: KpiCard;
  allocated_overhead_expenses: KpiCard;
  net_profit: KpiCard;
  pipeline_value: KpiCard;
  order_count: OrderCountCard;
  aov: KpiCard;
  time_series: TimeSeriesPoint[];
  diagnostic: DiagnosticInfo;
  // PART-1: informational shipping margin KPI. current=[] when both
  // shipping_revenue and shipping_np_cost are zero — frontend auto-hides.
  shipping_net: KpiCard;
}

File: backend/schemas/finance.py
"""
OrderHub CRM — Finance Schemas (FIN-1)

Pydantic models for the per-shop financial overview endpoint at
/api/shops/{shop_id}/finance. All money values are per-currency lists;
no FX conversion is performed.
"""

from typing import Literal

from pydantic import BaseModel


class CurrencyAmount(BaseModel):
    currency: str
    amount: float


class KpiCard(BaseModel):
    current: list[CurrencyAmount]
    previous: list[CurrencyAmount]
    # change_percent is computed over the primary currency (the one with
    # the largest current.amount). None when previous is zero or no data.
    change_percent: float | None


class OrderCountCard(BaseModel):
    current: int
    previous: int
    change_percent: float | None


class TimeSeriesPoint(BaseModel):
    date: str
    currency: str
    revenue: float
    net_profit: float


class DiagnosticInfo(BaseModel):
    orders_missing_cost: int
    total_orders_in_period: int


class ShopFinanceResponse(BaseModel):
    shop_id: str
    shop_name: str
    period_start_iso: str
    period_end_iso: str
    granularity: Literal["day", "month"]
    revenue: KpiCard
    cogs: KpiCard
    fees: KpiCard
    net_profit: KpiCard
    pipeline_value: KpiCard
    order_count: OrderCountCard
    aov: KpiCard
    time_series: list[TimeSeriesPoint]
    diagnostic: DiagnosticInfo
